Branch-SDK/src/io/branch/referral/BranchException.java

```diff
@@ -0,0 +1,49 @@
+package io.branch.referral;
+
+/**
+ * <p>
+ * Class for representing any Branch SDK exceptions
+ * Exception details are provided in exception message.
+ * </p>
+ */
+public class BranchException extends Exception {
```

Comment:
@qinweigong  It seems like BranchError.Java specified String are used only as parameters in callback methods and seems to be an ideal design.
In the case of getInstance() we can't return error messages and also we may have different type of errors.So Exception forces Developers to handle it in their own  way.What is your suggestion?